### PR TITLE
Fix usage of loop in a for else clause

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: tests/Fixtures/tags/for/loop_in_else.test [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+{% for item in [] %}
+{% else %}
+{{ loop.previous is null ? 'OK' : 'KO' }}
+{{ loop.next is null ? 'OK' : 'KO' }}
+{{ loop.first ? 'OK' : 'KO' }}
+{{ loop.last ? 'OK' : 'KO' }}
+{{ loop.index0 is same as 0 ? 'OK' : 'KO' }}
+{{ loop.revindex0 is same as 0 ? 'OK' : 'KO' }}
+{{ loop.length is same as 0 ? 'OK' : 'KO' }}
+{{ loop.depth is same as 1 ? 'OK' : 'KO' }}
```
Comment (fabpot) [inline tests/Fixtures/tags/for/loop_in_else.test:13]: Should be 0 maybe?

File: tests/Fixtures/tags/for/loop_in_else.test [hunk 2]
```diff
@@ -0,0 +1,29 @@
+--TEST--
+"for" tag loop behavior in an else clause
+--TEMPLATE--
+{% for iterator in iterators %}
+{% for item in iterator %}
+{% else %}
+{{ loop.previous is null ? 'OK' : 'KO' }}
+{{ loop.next is null ? 'OK' : 'KO' }}
+{{ loop.first ? 'OK' : 'KO' }}
```
Comment (fabpot) [inline tests/Fixtures/tags/for/loop_in_else.test:8]: Should be false?

File: tests/Fixtures/tags/for/loop_in_else.test [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+{{ loop.last ? 'OK' : 'KO' }}
```
Comment (fabpot) [inline tests/Fixtures/tags/for/loop_in_else.test:9]: Should be false?

File: tests/Fixtures/tags/for/loop_in_else.test (hunk 1 above)
Comment (stof) [reply tests/Fixtures/tags/for/loop_in_else.test:13]: to me, it is fine to be 1. This is the depth of the current loop (even if it was empty)

File: tests/Fixtures/tags/for/loop_in_else.test (hunk 2 above)
Comment (smnandre) [reply tests/Fixtures/tags/for/loop_in_else.test:8]: 👍  False seems more logical  to me (it's "not even" the first iteration)

File: tests/Fixtures/tags/for/loop_in_else.test (hunk 3 above)
Comment (smnandre) [reply tests/Fixtures/tags/for/loop_in_else.test:9]: 👍  False seems more logical  to me (it's not "an" iteration, so not the last one)

Comment (stof) [reply tests/Fixtures/tags/for/loop_in_else.test:9]: I would say that any usage of the `loop` variable inside the `else` block is non-sense anyway. Getting any of those boolean value would be fine IMO (a PHP error due to an uninitialized property in Twig is not)

**Pull request merged**